server/src/main/java/org/opensearch/search/query/QuerySearchResult.java

```diff
@@ -364,6 +365,7 @@ public void readFromWithId(ShardSearchContextId id, StreamInput in) throws IOExc
         nodeQueueSize = in.readInt();
         setShardSearchRequest(in.readOptionalWriteable(ShardSearchRequest::new));
         setRescoreDocIds(new RescoreDocIds(in));
+        tookTimeNanos = in.readVLong();
```

Comment:
Do we need to worry about backward compatibility?

Reply:
As in, if this function tries to read a serialized QuerySearchResult from before the change was made?

Reply:
Added version check for >= 3.0 for the read/write functions